modules/geolocation_log_to_kml.py

```diff
@@ -0,0 +1,116 @@
+"""
+Convert log file to KML file.
+"""
+
+import argparse
+import pathlib
+import re
+
+from .common.modules import location_global
+from .common.modules import location_local
+from .common.modules import position_global
+from .common.modules.kml import kml_conversion
+from .common.modules.mavlink import local_global_conversion
+
+
+DEFAULT_RESULTS_PATH = pathlib.Path("results")
+
+
+# pylint: disable=redefined-outer-name
+def convert_geolocation_log_to_kml(
+    log_file: str,
+    home_position: position_global.PositionGlobal,
+    document_name_prefix: str,
+    save_directory: str,
+) -> "tuple[bool, pathlib.Path | None]":
+    """
+    Given a geolocation log file with a specific format, return a corresponding KML file.
+
+    Args:
+        log_file (str): Path to the geolocation log file
+        document_name_prefix (str): Prefix name for saved KML file.
+        save_directory (str): Directory to save the KML file to.
+
+    Returns:
+        tuple[bool, pathlib.Path | None]: Returns (False, None) if function
+            failed to execute, otherwise (True, path) where path a pathlib.Path
+            object pointing to the KML file.
+    """
+    locations = []
+
+    with open(log_file, "r", encoding="utf-8") as f:
+        for line in f:
+            # find all the points within the line (should be just 1)
```

Comment:
Just 1? I thought there can be more than 1 detection?

Reply:
Hi @maxlou05, I'm currently trying to address this issue, but I am not given the format of the geolocation log files, so I don't know how many detections can be on this line.